### PR TITLE
[Merged by Bors] - feat: turn off bad simp lemmas

### PR DESCRIPTION
**Pull request by kim-em**
Removes some bad `simp` lemmas as identified by @mattrobball at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Infrastructure.20for.20tracking.20frequently.20applied.20simp.20theorems/near/459926416

### CODE REVIEW

**Review by TwoFX — APPROVED** [review]
bors r+

File: Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean [hunk 1]
```diff
@@ -640,6 +640,7 @@ lemma biproduct.whiskerEquiv_inv_eq_lift {f : J → C} {g : K → C} (e : J ≃
     · rintro rfl
       simp at h
 
+attribute [local simp] Sigma.forall in
```
Comment (mattrobball) [inline Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean:643]: Can this be inlined into the `simp` calls in the proof?

Comment (semorrison) [reply Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean:643]: I was too slow, but: this was used by an autoparam `aesop` call.

Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean:643]: I guess I didn't read carefully

**Pull request closed without merging**